lib/model.js

```diff
@@ -1412,6 +1412,13 @@ module.exports = (function() {
    * @return {Promise<undefined>}
    */
   Model.prototype.destroy = function(options) {
+    var self = this
+      , instances;
+
+    if (!options || !(options.where || options.truncate)) {
```

Comment:
I was just surprised to have have to change all calls to `Model.destroy()` to `Model.destroy({where: {}})`